### PR TITLE
DEV: Move `discourse.js` to `app/app.js`

### PR DESCRIPTION
**Pull request by eviltrout**
This is another thing to get our application in line with what Ember CLI
expects.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -324,7 +324,14 @@ def config.database_configuration
 
         ActionView::Base.precompiled_asset_checker = -> logical_path do
           default_checker[logical_path] ||
-            %w{qunit.js qunit.css test_helper.css test_helper.js wizard/test/test_helper.js}.include?(logical_path)
+            %w{qunit.js
```
Comment (jjaffeux) [inline config/application.rb:327]: I don't understand what is this file doing in this commit? how is this related?

Comment (eviltrout) [reply config/application.rb:327]: Sorry these errors came up while I was building this, but basically it means if there is a node_modules or dist anywhere in discourse everything would break. I can split it up if you prefer.

Comment (jjaffeux) [reply config/application.rb:327]: No up to you :)

Was just wondering if I was missing something :+1:
